pulser-core/pulser/pulse.py

```diff
@@ -103,15 +104,17 @@ def __init__(
             raise ValueError(
                 "The duration of detuning and amplitude waveforms must match."
             )
-        if np.any(amplitude.samples < 0):
+        if np.any(amplitude.samples.as_array(detach=True) < 0):
             raise ValueError(
                 "All samples of an amplitude waveform must be "
                 "greater than or equal to zero."
             )
         object.__setattr__(self, "amplitude", amplitude)
         object.__setattr__(self, "detuning", detuning)
-        phase = cast(float, phase)
-        object.__setattr__(self, "phase", float(phase) % (2 * np.pi))
+        assert not isinstance(phase, Parametrized)
```

Comment:
So phase can't be Parametrized ? If so, we should remove Parametrized from the possible types of phase.

Reply:
Resolved.